src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -863,5 +874,111 @@ std::shared_ptr<coreml::MLModelWrapper> drawing_classifier::export_to_coreml(
+                                               size_t version) {
+  auto model_iter = model_data.find("_model");
+  if (model_iter == model_data.end()) {
+    log_and_throw("The loaded turicreate model must contain '_model'!\n");
+  }
+  const flex_dict& model = variant_get_value<flex_dict>(model_iter->second);
+
+  // For a model trained on integer classes, when saved and loaded back,
+  // the classes are loaded as floats. The following code block casts
+  // the loaded "float" classes back to int.
+  if (model_data.count("classes")) {
+    flex_list classes_list =
+        variant_get_value<flex_list>(model_data.find("classes")->second);
+
+    if (classes_list.size() &&
+        classes_list.begin()->get_type() == flex_type_enum::FLOAT) {
+      flex_list new_classes_list;
+      new_classes_list.reserve(classes_list.size());
+
+      std::transform(classes_list.begin(), classes_list.end(),
+                     std::back_inserter(new_classes_list),
+                     [](flexible_type& ft) { return ft.to<flex_int>(); });
+
+      model_data.erase("classes");
+      model_data.emplace("classes", std::move(new_classes_list));
+
+    } else if (classes_list.empty()) {
+      log_and_throw(
+          "Error during loading model. 'classes' must contain at least one "
+          "class label.");
+    }
+  } else {
+    log_and_throw(
+        "Error during loading model. 'classes' not found in the saved model.");
+  }
+
+  flex_dict mxnet_data_dict;
+  flex_dict mxnet_shape_dict;
+
+  for (const auto& data : model) {
+    if (data.first == "data") {
+      mxnet_data_dict = data.second;
+    } else if (data.first == "shapes") {
+      mxnet_shape_dict = data.second;
+    }
+  }
+
+  auto cmp = [](const flex_dict::value_type& a,
+                const flex_dict::value_type& b) { return (a.first < b.first); };
+
+  std::sort(mxnet_data_dict.begin(), mxnet_data_dict.end(), cmp);
+  std::sort(mxnet_shape_dict.begin(), mxnet_shape_dict.end(), cmp);
+
+  float_array_map nn_params;
+
+  std::vector<float> layer_weight;
+  std::vector<size_t> layer_shape;
+
+  for (size_t ii = 0; ii < mxnet_data_dict.size(); ii++) {
+    std::string layer_name = mxnet_data_dict[ii].first;
+    flex_nd_vec mxnet_data_nd = mxnet_data_dict[ii].second.to<flex_nd_vec>();
+    flex_nd_vec mxnet_shape_nd = mxnet_shape_dict[ii].second.to<flex_nd_vec>();
+
+    const std::vector<double>& model_weight = mxnet_data_nd.elements();
+    const std::vector<double>& model_shape = mxnet_shape_nd.elements();
+    // load the weights
+    layer_weight.clear();
+    layer_shape.clear();
+    layer_weight.insert(layer_weight.end(), model_weight.begin(), model_weight.end());
+    layer_shape.insert(layer_shape.end(), model_shape.begin(), model_shape.end());
+
+    nn_params[layer_name] = shared_float_array::wrap(std::move(layer_weight),
+                                                     std::move(layer_shape));
+  }
+
+  // missing state from prior 6.0; use default value
```

Comment:
Why do we need to add these things? Are they required by any of the inference logic? I think this just misleads people. The point of the random seed for example is to allow people to create the model. But the random seed you're generating below doesn't actually help somebody reproduce the model you've loaded.

Reply:
because we need to call `init_model()`, which is required by that function call. The random state is just to make sure the call can be done successfully. The random state won't take any effect since weights will be overridden later.

Reply:
Yes but it will appear to the user in the python attributes and mislead them into thinking that it can be used to reproduce the model. Let's not put it in the state and instead make `init_model` more robust. Nothing should need random initialization for loading a saved model. The weights should be read from disk, not random. (Let's not pay the overhead of randomly initializing weights and then overwriting them, either.)

Reply:
good.

Reply:
addressed.

Reply:
True that. And I just want to make sure I understand the context here. For the loading saved model from the prior 6.0 version, we are not treating it as a checkpoint, which means the user can continue to train the model, but a completed model, where it can only be used for inferencing. Is my understanding correct?

Reply:
We do not support checkpointing at this time. No model we are loading from 5.8 or earlier will be a checkpoint.